### PR TITLE
Fix: page crash when trying to decode rep mint

### LINKED ISSUE #256: Page crashes when decoding a proposal that mints rep
**Description**
This bug was found by @MiltonTulli 

When trying to decode a proposal that mints rep, the page crashes

image

https://www.loom.com/share/3a087643ee704adb9741d1e91bd05acb

**Steps to Reproduce**

1. Go to a guild page
2. Click on a proposal that mints rep (like this one: [localhost link](http://localhost:3000/#/gnosis/0x3f842726188FcD932d43bcA291be28138228e6D9/proposal/0x8e9e391c2a06218ef16f4e2d1f2c001b18c8f4e68a92f8decd06bf5bb635a838))

**Expected behavior**
The app should not crash

### PR DESCRIPTION
**Pull request by dcrescimbeni**
# Description

Fixes bug causing a crash when trying to decode a rep mint action.

Closes #256 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Added unit test for the `shortenAddress` function.

# Checklist:

- [x] My code follows the existing style of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any UI changes have been tested and made responsive for mobile views

### CODE REVIEW

**Comment by rossneilson** [comment]
Does this error occur in production and qa, if so could we merge to release 3.1 as well?

**Comment by dcrescimbeni** [comment]
> Does this error occur in production and qa, if so could we merge to release 3.1 as well?

Yes, I think it's crashing also.

I'll need to copy the changes to the 3.1 branch since this PR has all the gov 1.5 changes.

**Comment by rossneilson** [comment]
Actually seems it is working in v3.1.0 so should be fine I think. Could you test in case I am missing something?

**Comment by dcrescimbeni** [comment]
TLDR: we shouldn't merge this into 3.1 as doing it will crash the page.

The crash was caused because the component was expecting a `to` argument to decode the call, but the argument is now named `account` in the contract.

In the v2.0 branch of the contracts, the ERC20SnapshotRep `mint` function arguments were renamed. The address was named `to` and now it's named `account`.

So it's working on the 3.1 branch because both the ABI and the component have the same name, and was crashing in the local env because the component was expecting a `to` parameter, but the ABI was returning an `account` param.

If we merge this, the component will expect a `account` param, but the ABI will return a `to` param. Good thing is, it will not crash the page, but will return `null`.

Since in the new version we'll be using the v2 ABI, this fix should only be merged for the new release.

**Review by rulfo71 — APPROVED** [review]
looks good!

**Comment by rossneilson** [comment]
> In the v2.0 branch of the contracts, the ERC20SnapshotRep `mint` function arguments were renamed. The address was named `to` and now it's named `account`.

I think we need to think a bit about how we do version control in DAVI. This contract change would be a breaking change to the interface (admittedly a small one) and we need to continue to support te deployed versions of guilds.

My opinion for now would be to document this change but continue to use our old contract versions for guilds in DAVI. We can add this contract change when we do a larger version increment on guilds (Guilds v2).

**Comment by dcrescimbeni** [comment]
> I think we need to think a bit about how we do version control in DAVI. This contract change would be a breaking change to the interface (admittedly a small one) and we need to continue to support te deployed versions of guilds.

This change in particular won't affect it since the decoded variables it's just a human thing, and those don't matter at the contract level.

But a thing we will face is the guilds upgrade. It adds some new methods and events. Right now both ABIs have the same name.

So, it's complicated I guess. If we don't merge this, combined with the way the current dev-scripts work, it will break production in the next release.

For now, I think the most reasonable way is to merge it in develop, and then I agree that we need to think about how do we do version control with guilds.
